src/Compiler/AbstractIL/il.fs

```diff
@@ -433,8 +433,8 @@ let AssemblyRefUniqueStampGenerator = UniqueStampGenerator<AssemblyRefData>()
 type ILAssemblyRef(data) =
     let pkToken key =
         match key with
-        | Some (PublicKey bytes) -> Some(PublicKey(SHA1.sha1HashBytes bytes))
-        | Some (PublicKeyToken token) -> Some(PublicKey token)
+        | Some(PublicKey bytes) -> Some(PublicKey(SHA1.sha1HashBytes bytes))
+        | Some(PublicKeyToken token) -> Some(PublicKey token)
```

Comment:
I hate this with a passion. Does the style guide really say there shouldn't be a space between a long ident and arg pats?

Reply:
Hey @kerams,
I think the background to this can be found [here](https://github.com/fsharp/fslang-design/issues/712)

Reply:
@kerams I think I can try to share some of the context here.

Consider these cases:
```fsharp
module M =
    let f1 a b = ()
    f1 1 2

    let f2 (a, b) = ()
    f2 (1, 2)


type T() =
    static member _.Method(a, b) = ()

T.Method(1, 2)
```

The `f1 1 2`, `f2 (1, 2)` and `T.Method(1, 2)` are seemingly the way that the most people format their F# code, and Fantomas team tried to come up with logic that would try to preserve the most common styles.

The problem for the tooling (including Fantomas) is there's no difference between `f2 (1, 2)`, `M.f2 (1, 2)` and `T.Method(1, 2)` in the syntax tree, and this is inherited from the language design. As things stand now, there's a heuristic that chooses between the style depending on the name of the 'function' expression: functions normally begin with a lowercase letter, and methods begin with an uppercase one. Even though I really dislike the idea on depending on names for formatting, we couldn't find any better approach to tackle this. There're some rare cases where it doesn't work as expected (e.g. there're unfortunately a few functions starting with an uppercase letter, or somebody can even write a curried method), but with all the downsides it's still the best solution we've had for this problem to date.

One thing was missing for quite some time, though: patterns formatting didn't use the same heuristic and was formatted inconsistently with expressions. I'm really glad we're making things more consistent now here.

Reply:
See, I don't agree with making a distinction between methods and functions with tupled arguments, regardless of casing. Do you need to use a call chain? Drop the space, otherwise always include it.

Reply:
> See, I don't agree with making a distinction between methods and functions with tupled arguments, regardless of casing.

Yes, I understand and I also don't agree with some of the other choices. I just tried to explain that in this particular case, like in many others, the intention was to try to preserve the most common styles, which probably is the most important thing for a formatter, and patterns were overlooked when that was made.